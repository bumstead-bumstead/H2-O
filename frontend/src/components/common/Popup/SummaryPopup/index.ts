export { default as SummaryPopup } from './SummaryPopup';
