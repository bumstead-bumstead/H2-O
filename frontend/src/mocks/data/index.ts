export * from './trims';
export * from './technical-spec';
export * from './selectionInfo';
export * from './priceRange';
export * from './model-type';
export * from './interiors';
