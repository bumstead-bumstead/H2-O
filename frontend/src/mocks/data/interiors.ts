export const interiors = [
  {
    id: 1,
    name: '퀼팅 천연(블랙)',
    choiceRatio: 62,
    price: 0,
    fabricImage: 'https://h2o-static-contents.s3.ap-northeast-2.amazonaws.com/internal/colorchip-interior-black.png',
    bannerImage: 'https://h2o-static-contents.s3.ap-northeast-2.amazonaws.com/internal/img-interior-black.png',
  },
  {
    id: 2,
    name: '쿨그레이',
    choiceRatio: 38,
    price: 0,
    fabricImage: 'https://h2o-static-contents.s3.ap-northeast-2.amazonaws.com/internal/colorchip-interior-gray.png',
    bannerImage: 'https://h2o-static-contents.s3.ap-northeast-2.amazonaws.com/internal/img-interior-gray.png',
  },
];
