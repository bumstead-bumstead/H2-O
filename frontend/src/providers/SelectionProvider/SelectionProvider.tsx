import { Dispatch, PropsWithChildren, createContext, useMemo } from 'react';
import useSelectionReducer, { type Action, type State } from './hooks/useSelectionReducer';

interface SelectionContextType {
  selectionInfo: State;
  totalPrice: number;
  dispatch: Dispatch<Action>;
}

export const SelectionContext = createContext<SelectionContextType | null>(null);

function SelectionProvider({ children }: PropsWithChildren) {
  const [state, dispatch] = useSelectionReducer();

  const totalPrice = useMemo(() => {
    const { trim, powerTrain, bodyType, driveTrain, exteriorColor, interiorColor, extraOptions } = state;
    const options = [trim, powerTrain, bodyType, driveTrain, exteriorColor, interiorColor, extraOptions];

    return options.reduce((acc, curr) => acc + (curr?.price || 0), 0);
  }, [state]);

  const contextValue = useMemo(
    () => ({
      selectionInfo: state,
      totalPrice,
      dispatch,
    }),
    [state, totalPrice],
  );

  return <SelectionContext.Provider value={contextValue}>{children}</SelectionContext.Provider>;
}

export default SelectionProvider;
