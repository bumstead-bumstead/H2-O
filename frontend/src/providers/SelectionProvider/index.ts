export { default as SelectionProvider, SelectionContext } from './SelectionProvider';
export type * from './types';
