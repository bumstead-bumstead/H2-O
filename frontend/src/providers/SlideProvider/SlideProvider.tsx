import type { Dispatch, PropsWithChildren, SetStateAction } from 'react';
import { createContext, useState } from 'react';

export interface SlideContextType {
  currentSlide: number;
  setCurrentSlide: Dispatch<SetStateAction<number>>;
}

export const SlideContext = createContext<SlideContextType | null>(null);

function SlideProvider({ children }: PropsWithChildren) {
  const [currentSlide, setCurrentSlide] = useState(0);

  return <SlideContext.Provider value={{ currentSlide, setCurrentSlide }}>{children}</SlideContext.Provider>;
}

export default SlideProvider;
