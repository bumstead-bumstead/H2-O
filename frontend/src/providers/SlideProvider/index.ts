export { default as SlideProvider, SlideContext } from './SlideProvider';
