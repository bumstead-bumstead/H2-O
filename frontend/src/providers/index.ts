export { StyleProvider } from './StyleProvider';
export { SlideProvider } from './SlideProvider';
export { SelectionProvider } from './SelectionProvider';
