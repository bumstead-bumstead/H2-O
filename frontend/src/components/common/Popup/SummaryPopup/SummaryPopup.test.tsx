import { PropsWithChildren, ReactElement } from 'react';
import { render } from '@testing-library/react';
import SummaryPopup from './SummaryPopup';
import { calculateTotalPrice, selectionInfo } from '@/mocks/data';
import { render as customRender, fireEvent, screen, waitFor } from '@/tests/test-util';
import { StyleProvider } from '@/providers';
import { SelectionContext } from '@/providers/SelectionProvider';
import { SlideContext } from '@/providers/SlideProvider';

const dispatchSelectionInfo = vi.fn();
const dispatchSlideInfo = vi.fn();

function TestProviders({ children }: PropsWithChildren) {
  return (
    <StyleProvider>
      <SlideContext.Provider value={{ currentSlide: 0, setCurrentSlide: dispatchSlideInfo }}>
        <SelectionContext.Provider
          value={{ selectionInfo, totalPrice: calculateTotalPrice(selectionInfo), dispatch: dispatchSelectionInfo }}
        >
          {children}
        </SelectionContext.Provider>
      </SlideContext.Provider>
    </StyleProvider>
  );
}

function TestProvidersWithoutExtraOptions({ children }: PropsWithChildren) {
  const newSelectionInfo = { ...selectionInfo, extraOptions: { optionList: [], price: 0 } };
  return (
    <StyleProvider>
      <SlideContext.Provider value={{ currentSlide: 0, setCurrentSlide: dispatchSlideInfo }}>
        <SelectionContext.Provider
          value={{
            selectionInfo: newSelectionInfo,
            totalPrice: calculateTotalPrice(newSelectionInfo),
            dispatch: dispatchSelectionInfo,
          }}
        >
          {children}
        </SelectionContext.Provider>
      </SlideContext.Provider>
    </StyleProvider>
  );
}

function testRender(ui: ReactElement, isExtraOption = true, options = {}) {
  isExtraOption
    ? render(ui, { wrapper: TestProviders, ...options })
    : render(ui, { wrapper: TestProvidersWithoutExtraOptions, ...options });
}

describe('SummaryPopup 컴포넌트 테스트', () => {
  beforeEach(() => {
    const portalRoot = document.createElement('div');
    portalRoot.setAttribute('id', 'portal-root');

    document.body.appendChild(portalRoot);
  });

  const handleClickCloseButton = vi.fn();

  it('SelectionInfo가 없으면 로딩 UI가 렌더링된다.', () => {
    customRender(<SummaryPopup handleClickCloseButton={handleClickCloseButton} />);

    expect(screen.getByText('데이터를 불러오는 중입니다...')).toBeInTheDocument();
  });

  it('SelectionInfo가 있으면 SummaryPopup 컴포넌트가 렌더링된다.', () => {
    testRender(<SummaryPopup handleClickCloseButton={handleClickCloseButton} />);

    expect(screen.getByTestId('summary-modal')).toBeInTheDocument();
  });

  it('SummaryPopup 컴포넌트의 cancel 아이콘을 누르면 handleClickCloseButton이 실행된다.', () => {
    const handleClickCloseButton = vi.fn();
    testRender(<SummaryPopup handleClickCloseButton={handleClickCloseButton} />);

    const cancelIcon = screen.getByTestId('Cancel');
    fireEvent.click(cancelIcon);

    expect(handleClickCloseButton).toBeCalledTimes(1);
  });

  it('SummaryPopup의 dimmed 영역을 클릭하면 handleClickCloseButton이 실행된다.', () => {
    const handleClickCloseButton = vi.fn();
    testRender(<SummaryPopup handleClickCloseButton={handleClickCloseButton} />);

    const dimmed = screen.getByTestId('dimmed');
    fireEvent.click(dimmed);

    expect(handleClickCloseButton).toBeCalledTimes(1);
  });

  it('SummaryPopup의 견적 완료하기 버튼을 클릭하면 handleClickCloseButton와 setCurrentSlider가 실행된다.', () => {
    const handleClickCloseButton = vi.fn();
    testRender(<SummaryPopup handleClickCloseButton={handleClickCloseButton} />);

    const button = screen.getByText('견적 완료하기');
    fireEvent.click(button);

    expect(handleClickCloseButton).toBeCalledTimes(1);
    expect(dispatchSlideInfo).toBeCalledTimes(1);
  });

  it('SummaryPopup에 3개의 divider로 분리되어 선택사항에 대한 정보가 표시된다.', () => {
    testRender(<SummaryPopup handleClickCloseButton={handleClickCloseButton} />);

    expect(screen.getAllByTestId('divider')).toHaveLength(3);
  });

  it('extraOptions의 optionList가 빈 배열이면 - 를 표시한다.', () => {
    testRender(<SummaryPopup handleClickCloseButton={handleClickCloseButton} />, false);

    expect(screen.getAllByText('-')).toHaveLength(1);
  });

  it('내장 토글 버튼을 클릭하면 이미지가 외장색상에서 내장색상으로 바뀐다.', async () => {
    testRender(<SummaryPopup handleClickCloseButton={handleClickCloseButton} />);
    expect(screen.getByAltText('외장색상 이미지')).toBeInTheDocument();

    const toggleButton = screen.getByText('내장');
    fireEvent.click(toggleButton);

    await waitFor(() => {
      expect(screen.getByAltText('내장색상 이미지')).toBeInTheDocument();
    });
  });

  it('SummaryPopup 스냅샷 테스트', () => {
    testRender(<SummaryPopup handleClickCloseButton={handleClickCloseButton} />);

    expect(screen.getByTestId('summary-modal')).toMatchSnapshot();
  });
});
