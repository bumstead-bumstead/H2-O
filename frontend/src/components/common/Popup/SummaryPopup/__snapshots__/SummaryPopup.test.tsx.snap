// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SummaryPopup 컴포넌트 테스트 > SummaryPopup 스냅샷 테스트 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  width: 100%;
  height: 100%;
}

.emotion-1 {
  font-family: Hyundai Sans Head KR,sans-serif;
  font-size: 18px;
  font-weight: 500;
  line-height: 28px;
  letter-spacing: -0.03em;
  color: #343434;
  width: 100%;
  height: 53px;
  text-align: center;
  padding-top: 20px;
}

.emotion-2 {
  position: absolute;
  right: 20px;
  top: 21px;
  cursor: pointer;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  gap: 32px;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 62px;
  margin-left: 40px;
  gap: 20px;
}

.emotion-5 {
  width: 418px;
  height: 212px;
  object-fit: cover;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  background-color: #FFFFFF;
  border: 1px solid #EDEDEE;
  border-radius: 22px;
  width: 154px;
  height: 44px;
  pointer-events: none;
}

.emotion-7 {
  display: none;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  overflow: hidden;
  cursor: pointer;
}

.emotion-9 {
  position: absolute;
  left: 0px;
  background: #00397B;
  border-radius: 20px;
  width: 72px;
  height: 36px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: Hyundai Sans Text KR,sans-serif;
  font-size: 14px;
  font-weight: 500;
  line-height: 22px;
  letter-spacing: -0.03em;
  color: #F4F5F7;
  width: 72px;
  height: 36px;
  z-index: 1;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  pointer-events: none;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: Hyundai Sans Text KR,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 22px;
  letter-spacing: -0.03em;
  color: #B6B8BB;
  width: 72px;
  height: 36px;
  z-index: 1;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  pointer-events: fill;
}

.emotion-12 {
  width: 324px;
  height: 304px;
  margin-right: 36px;
  overflow: scroll;
}

.emotion-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 8px;
  width: 100%;
  height: 22px;
}

.emotion-14 {
  color: #A3A6AA;
  font-family: Hyundai Sans Text KR,sans-serif;
  font-size: 12px;
  font-weight: 400;
  line-height: 18px;
  letter-spacing: -0.03em;
  width: 56px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
}

.emotion-16 {
  color: #212122;
  font-family: Hyundai Sans Text KR,sans-serif;
  font-size: 12px;
  font-weight: 500;
  line-height: 18px;
  letter-spacing: -0.03em;
  width: 160px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.emotion-17 {
  color: #212122;
  font-family: Hyundai Sans Text KR,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 22px;
  letter-spacing: -0.03em;
}

.emotion-23 {
  width: 100%;
  height: 1px;
  margin: 16px 0px;
  background-color: #F4F5F7;
}

.emotion-61 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
  font-family: Hyundai Sans Text KR,sans-serif;
  font-size: 12px;
  font-weight: 400;
  line-height: 18px;
  letter-spacing: -0.03em;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #626466;
  width: 100%;
  height: 64px;
  padding: 8px 36px 24px 0px;
  gap: 8px;
}

.emotion-62 {
  font-family: Hyundai Sans Head KR,sans-serif;
  font-size: 24px;
  font-weight: 500;
  line-height: 32px;
  letter-spacing: -0.03em;
  color: #00397B;
}

.emotion-63 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: Hyundai Sans Text KR,sans-serif;
  font-size: 16px;
  font-weight: 500;
  line-height: 18px;
  letter-spacing: -0.03em;
  background-color: #00224A;
  color: #FFFFFF;
  border-radius: 4px;
  width: 850px;
  height: 52px;
}

.emotion-63:hover {
  background-color: #00397B;
}

.emotion-63:active {
  background-color: #001731;
}

.emotion-63:disabled {
  background-color: #C8CACC;
  cursor: not-allowed;
}

<div
  class="emotion-0"
  data-testid="summary-modal"
>
  <div
    class="emotion-1"
  >
    견적요약
    <svg
      class="emotion-2"
      data-testid="Cancel"
      fill="#414344"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"
        fill="inherit"
      />
    </svg>
  </div>
  <div
    class="emotion-3"
  >
    <div
      class="emotion-4"
    >
      <img
        alt="외장색상 이미지"
        class="emotion-5"
        src="https://h2o-static-contents.s3.ap-northeast-2.amazonaws.com/trim/le-blanc/blue_exterior/image_001.png"
      />
      <div
        class="emotion-6"
      >
        <input
          class="emotion-7"
          id=":r7:"
          type="checkbox"
        />
        <label
          class="emotion-8"
          for=":r7:"
        >
          <span
            class="emotion-9"
          />
          <div
            class="emotion-10"
          >
            외장
          </div>
          <div
            class="emotion-11"
          >
            내장
          </div>
        </label>
      </div>
    </div>
    <div
      class="emotion-12"
    >
      <div
        class="emotion-13"
        height="22px"
        width="100%"
      >
        <span
          class="emotion-14"
          color="gray500"
        >
          모델
        </span>
        <div
          class="emotion-15"
          height="100%"
          width="100%"
        >
          <span
            class="emotion-16"
            color="gray900"
          >
            팰리세이드
          </span>
          <span
            class="emotion-17"
            color="gray900"
          >
            +0
            원
          </span>
        </div>
      </div>
      <div
        class="emotion-13"
        height="22px"
        width="100%"
      >
        <span
          class="emotion-14"
          color="gray500"
        >
          트림
        </span>
        <div
          class="emotion-15"
          height="100%"
          width="100%"
        >
          <span
            class="emotion-16"
            color="gray900"
          >
            Le Blanc
          </span>
          <span
            class="emotion-17"
            color="gray900"
          >
            +38,960,000
            원
          </span>
        </div>
      </div>
      <div
        class="emotion-23"
        data-testid="divider"
      />
      <div
        class="emotion-13"
        height="22px"
        width="100%"
      >
        <span
          class="emotion-14"
          color="gray500"
        >
          파워트레인
        </span>
        <div
          class="emotion-15"
          height="100%"
          width="100%"
        >
          <span
            class="emotion-16"
            color="gray900"
          >
            디젤 2.2
          </span>
          <span
            class="emotion-17"
            color="gray900"
          >
            +1,480,000
            원
          </span>
        </div>
      </div>
      <div
        class="emotion-13"
        height="22px"
        width="100%"
      >
        <span
          class="emotion-14"
          color="gray500"
        >
          바디타입
        </span>
        <div
          class="emotion-15"
          height="100%"
          width="100%"
        >
          <span
            class="emotion-16"
            color="gray900"
          >
            7인승
          </span>
          <span
            class="emotion-17"
            color="gray900"
          >
            +0
            원
          </span>
        </div>
      </div>
      <div
        class="emotion-13"
        height="22px"
        width="100%"
      >
        <span
          class="emotion-14"
          color="gray500"
        >
          구동방식
        </span>
        <div
          class="emotion-15"
          height="100%"
          width="100%"
        >
          <span
            class="emotion-16"
            color="gray900"
          >
            2WD
          </span>
          <span
            class="emotion-17"
            color="gray900"
          >
            +0
            원
          </span>
        </div>
      </div>
      <div
        class="emotion-23"
        data-testid="divider"
      />
      <div
        class="emotion-13"
        height="22px"
        width="100%"
      >
        <span
          class="emotion-14"
          color="gray500"
        >
          외장색상
        </span>
        <div
          class="emotion-15"
          height="100%"
          width="100%"
        >
          <span
            class="emotion-16"
            color="gray900"
          >
            문라이트 블루 펄
          </span>
          <span
            class="emotion-17"
            color="gray900"
          >
            +0
            원
          </span>
        </div>
      </div>
      <div
        class="emotion-13"
        height="22px"
        width="100%"
      >
        <span
          class="emotion-14"
          color="gray500"
        >
          내장색상
        </span>
        <div
          class="emotion-15"
          height="100%"
          width="100%"
        >
          <span
            class="emotion-16"
            color="gray900"
          >
            퀼팅 천연(블랙)
          </span>
          <span
            class="emotion-17"
            color="gray900"
          >
            +0
            원
          </span>
        </div>
      </div>
      <div
        class="emotion-23"
        data-testid="divider"
      />
      <div
        class="emotion-13"
        height="22px"
        width="100%"
      >
        <span
          class="emotion-14"
          color="gray500"
        >
          옵션
        </span>
        <div
          class="emotion-15"
          height="100%"
          width="100%"
        >
          <span
            class="emotion-16"
            color="gray900"
          >
            컴포트 Ⅱ
          </span>
          <span
            class="emotion-17"
            color="gray900"
          >
            +1,090,000
            원
          </span>
        </div>
      </div>
      <div
        class="emotion-13"
        height="22px"
        width="100%"
      >
        <span
          class="emotion-14"
          color="gray500"
        >
          옵션
        </span>
        <div
          class="emotion-15"
          height="100%"
          width="100%"
        >
          <span
            class="emotion-16"
            color="gray900"
          >
            빌트인 캠(보조배터리 포함)
          </span>
          <span
            class="emotion-17"
            color="gray900"
          >
            +690,000
            원
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="emotion-61"
  >
    <span>
      현재 총 가격
    </span>
    <div
      class="emotion-62"
    >
      42,220,000
       원
    </div>
  </div>
  <button
    class="emotion-63"
  >
    견적 완료하기
  </button>
</div>
`;
